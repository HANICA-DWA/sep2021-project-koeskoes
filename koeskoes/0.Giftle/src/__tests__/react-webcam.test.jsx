import * as React from "react";
import * as renderer from "react-test-renderer";
import Webcam from "react-webcam";

test("webcam renders correctly", () => {
  const tree = renderer.create(
    <Webcam
      audio={false}
      className="react-webcam"
      imageSmoothing={false}
      minScreenshotHeight={1000}
      minScreenshotWidth={1000}
      onUserMedia={() => {}}
      onUserMediaError={() => {}}
      style={{ transform: "rotate(180deg)" }}
      videoConstraints={{
        width: 160,
        height: 120,
        frameRate: 15,
      }}
      height={1000}
      width={1000}
    />
  );

  expect(tree.toJSON()).toMatchSnapshot();
});

test("sets <video/> muted to false when props.audio is true for recording audio", () => {
  const tree = renderer.create(
    <Webcam
      audio={true}
      className="react-webcam"
      onUserMedia={() => {}}
      onUserMediaError={() => {}}
      style={{ transform: "rotate(180deg)" }}
      videoConstraints={{
        width: 160,
        height: 120,
        frameRate: 15,
      }}
      height={1000}
      width={1000}
    />
  );

  expect(tree.root.findByType("video").props.muted).toBe(false);
});

test("audio recording is muted", () => {
  const tree = renderer.create(
    <Webcam
      audio={false}
      className="react-webcam"
      onUserMedia={() => {}}
      onUserMediaError={() => {}}
      style={{ transform: "rotate(180deg)" }}
      videoConstraints={{
        width: 160,
        height: 120,
        frameRate: 15,
      }}
      height={1000}
      width={1000}
    />
  );

  expect(tree.root.findByType("video").props.muted).toBe(true);
});
