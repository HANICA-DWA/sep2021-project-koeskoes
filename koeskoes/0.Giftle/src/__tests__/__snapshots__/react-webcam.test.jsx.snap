// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webcam renders correctly 1`] = `
<video
  autoPlay={true}
  className="react-webcam"
  height={1000}
  muted={true}
  playsInline={true}
  style={
    Object {
      "transform": "rotate(180deg)",
    }
  }
  width={1000}
/>
`;
